packages/hdwallet-native/src/adapter.ts

```diff
@@ -1,49 +1,48 @@
 import * as core from "@shapeshiftoss/hdwallet-core";
-import { create } from "./native";
+import { create, NativeHDWallet } from "./native";
 
-type NativeAdapterArgs = {
-  mnemonic: string;
+export type NativeAdapterArgs = {
+  mnemonic?: string;
   deviceId: string;
 };
 
 export class NativeAdapter {
   keyring: core.Keyring;
   deviceId: string;
-
-  private mnemonic: string;
+  #mnemonic: string;
 
   private constructor(keyring: core.Keyring, args: NativeAdapterArgs) {
     this.keyring = keyring;
-    this.mnemonic = args.mnemonic;
     this.deviceId = args.deviceId;
+    this.#mnemonic = args.mnemonic;
   }
 
   static useKeyring(keyring: core.Keyring, args: NativeAdapterArgs) {
     return new NativeAdapter(keyring, args);
   }
 
   async initialize(): Promise<number> {
-    let wallet = this.keyring.get(this.deviceId);
+    let wallet = this.keyring.get<NativeHDWallet>(this.deviceId);
 
     if (!wallet) {
-      wallet = create(this.mnemonic, this.deviceId);
+      wallet = create({ mnemonic: this.#mnemonic, deviceId: this.deviceId });
       this.keyring.add(wallet, this.deviceId);
+      this.keyring.decorateEvents(this.deviceId, wallet.events);
     }
 
-    wallet.initialize();
+    if (this.#mnemonic) {
```

Comment:
We don't really need to wrap this is the mnemonic check now with the addition of `needsMnemonic` in `wallet.initialize`. Thoughts?

Reply:
that's actually WHY we need it, because otherwise it'll immediate raise the mnemonic required event immediately on initializing the adapter, which is what I didn't want to happen.

Reply:
BTW, in my head I'm considering "adapter.initialize()" as a separate function from "wallet.initialize". I see the former as setting up the adapter and the latter as setting up the wallet. In our case, setting up the adapter itself shouldn't require the mnemonic, only initializing the wallet itself should. Maybe I'm thinking about that wrong, or not how it was intended?

Reply:
No, you are good, I confused myself. You are correct